darwin/Platform.c

```diff
@@ -725,3 +725,58 @@ void Platform_gettime_monotonic(uint64_t* msec) {
 #endif
 
 }
+
+#ifndef OSRELEASEFILE
+#define OSRELEASEFILE "/System/Library/CoreServices/SystemVersion.plist"
+#endif
```

Comment:
Isn't this always defined in `config.h` through `configure`? And if not, this code should FTBTS or at least properly fallback to "No OS release" if `--without-os-release` was specified for `configure`.

Reply:
@natoscott My intent is to remove user confusion when the `--with-os-release` option is there. If you do need an override, `CFLAGS='-DOSRELEASEFILE="/foo/os-release"'` is still available (I'm not suggesting the removal of that macro token).

The configure option is badly designed - first it didn't handle what to do `--without-os-release`, second it checks the file in the build machine even when cross compiling (one of the issues that I tried to fix in #1808). Needless to say that the paths `/etc/os-release` and `/usr/lib/os-release` are meant to be hard-coded according to the spec. Distros should not use other locations for the file.

Reply:
OK, I agree - I've added a follow-up commit that removes the configure option entirely.